scrape/target.go

```diff
@@ -422,41 +399,54 @@ func populateLabels(lset labels.Labels, cfg *config.ScrapeConfig) (res, orig lab
 		lb.Set(model.InstanceLabel, addr)
 	}
 
-	res = lb.Labels()
+	res := lb.Labels()
 	for _, l := range res {
 		// Check label values are valid, drop the target if not.
 		if !model.LabelValue(l.Value).IsValid() {
-			return nil, nil, errors.Errorf("invalid label value for %q: %q", l.Name, l.Value)
+			return nil, errors.Errorf("invalid label value for %q: %q", l.Name, l.Value)
 		}
 	}
-	return res, preRelabelLabels, nil
+	return res, nil
 }
 
-// targetsFromGroup builds targets based on the given TargetGroup and config.
-func targetsFromGroup(tg *targetgroup.Group, cfg *config.ScrapeConfig) ([]*Target, error) {
-	targets := make([]*Target, 0, len(tg.Targets))
+// stackLabels stacks labels from different sources for a particular target.
+// Sequence of adding labels is very important.
```

Comment:
I don't think we need this here, it's duplicating what the code says, and is confusing as even me knowing the code had to think a bit - as this is using terms that we don't use anywhere. If someone breaks this, we'll know from the unittests.

This is also not what the current logic does. Query parameters supersede SD labels. We can't break anything here, as users may depend on it.